Sources/XiEditor/RPCSending.swift

```diff
@@ -222,126 +257,126 @@ class StdoutRPCSender: RPCSending {
     }
 
     private func handleRequest(json: [String: AnyObject]) {
-        guard let method = json["method"] as? String, let params = json["params"], let id = json["id"]
+        guard let jsonMethod = json["method"] as? String,
+              let params = json["params"],
+              let id = json["id"],
+              let method = RPCRequestMethod(rawValue: jsonMethod),
+              let args = params as? [[String: AnyObject]],
+              let result = client?.measureWidth(args: args)
             else {
-                print("unknown json from core: \(json)")
+                assertionFailure("unknown json from core: \(json)")
                 return
         }
 
         switch method {
-        case "measure_width":
-            let args = params as! [[String: AnyObject]]
-            if let result = self.client?.measureWidth(args: args) {
-                self.sendResult(id: id, result: result)
-            }
-        default:
-            print("unknown request \(method)")
+        case .measureWidth:
+            sendResult(id: id, result: result)
         }
     }
 
     private func handleNotification(json: [String: AnyObject]) {
-        guard let method = json["method"] as? String, let params = json["params"]
+        guard let jsonMethod = json["method"] as? String,
+              let params = json["params"],
+              let method = RPCNotificationMethod(rawValue: jsonMethod)
             else {
-                print("unknown json from core: \(json)")
+                assertionFailure("unknown json from core: \(json)")
                 return
         }
-        let viewIdentifier = params["view_id"] as? ViewIdentifier
 
+        let viewIdentifier = params["view_id"] as? ViewIdentifier ?? ""
```

Comment:
so this really comes down to the implicit contracts that exist between core and client. We're responsible for both of these components; this isn't untrusted input. If this param is missing, that represents core breaking the API contract, and if that's the case there's really not a lot of value in continuing to run. It's probably even _bad_ to continue to run, because we're hiding a fundamental problem; crashing in an informative way is going to probably help us more.

Reply:
yup, makes total sense! undid it already! 👍